datacube/storage/masking.py

```diff
@@ -112,7 +112,7 @@ def valid_data_mask(data):
     :return: Dataset or DataArray
     """
     if isinstance(data, Dataset):
-        return data.apply(valid_data_mask)
+        return data.map(valid_data_mask)
```

Comment:
This will fail in versions of `xarray` older than v0.14.1 (19 Nov 2019), probably not what we want.

It's annoying, but we need to choose to:
- suffer the deprecation warning
- add an ugly version check here
- start requiring xarray >= 0.14.1

Reply:
so they just introduced `.map` and deprecated `.apply` in the same version? awesome. reverted that commit.